examples/tensorflow-profiling-examples/trainer-hooks-ext/model.py

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/env python
+
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+
+import tensorflow as tf
+from tensorflow.python.training.basic_session_run_hooks import _HookTimer
+
+tf.logging.set_verbosity(tf.logging.INFO)
+
+BUCKET = None  # set from task.py
+PATTERN = 'of'  # gets all files
+TRAIN_STEPS = 10000
+CSV_COLUMNS = 'weight_pounds,is_male,mother_age,plurality,gestation_weeks,key'.split(',')  # noqa: E501
+LABEL_COLUMN = 'weight_pounds'
+KEY_COLUMN = 'key'
+DEFAULTS = [[0.0], ['null'], [0.0], ['null'], [0.0], ['nokey']]
+
+# Define some hyperparameters
+BATCH_SIZE = 512
+N_EMBEDS = 3
+FIRST_LAYER_SIZE = 64
+NUM_LAYERS = 3
+DROPOUT_RATE = 0.1
+LEARNING_RATE = 0.001
+EVAL_INTERVAL = 200
+
+TRANSFORMED_TRAIN_DATA_FILEBASE = 'train_transformed'
+TRANSFORMED_TEST_DATA_FILEBASE = 'test_transformed'
+EXPORTED_MODEL_DIR = 'exported_model_dir'
+
+
+class LambdaTriggerTimer(_HookTimer):
```

Comment:
_HookTimer is a private class to tensorflow, do you absolutely need to inherit it here?
You aren't calling any of the _HookTimer methods, I don't think you need it?

Reply:
yep, that's a bad pattern. I'm calling __init__() now.
it looks to be more consistent since tf.train.ProfilerHook expects _HookTimer